packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -1,482 +1,9 @@
-        if (this.state.selectedDate && formatDate && formatDate(this.state.selectedDate) === inputValue) {
-          date = this.state.selectedDate;
-        } else {
-          date = parseDateFromString!(inputValue);
-
-          // Check if date is null, or date is Invalid Date
-          if (!date || isNaN(date.getTime())) {
-            // Reset invalid input field, if formatting is available
-            if (formatDate) {
-              date = this.state.selectedDate;
-              this.setState({
-                formattedDate: formatDate(date!).toString()
-              });
-            }
-
-            this.setState({
-              errorMessage: strings!.invalidInputErrorMessage || ' '
-            });
-          } else {
-            // Check against optional date boundaries
-            if (this._isDateOutOfBounds(date, minDate, maxDate)) {
-              this.setState({
-                errorMessage: strings!.isOutOfBoundsErrorMessage || ' '
-              });
-            } else {
-              this.setState({
-                selectedDate: date,
-                errorMessage: ''
-              });
-
-              // When formatting is available. If formatted date is valid, but is different from input, update with formatted date
-              // This occurs when an invalid date is entered twice
-              if (formatDate && formatDate(date) !== inputValue) {
-                this.setState({
-                  formattedDate: formatDate(date).toString()
-                });
-              }
-            }
-          }
-        }
-      } else {
-        // No input date string shouldn't be an error if field is not required
-        this.setState({
-          errorMessage: ''
-        });
-      }
-
-      // Execute onSelectDate callback
-      if (onSelectDate) {
-        // If no input date string or input date string is invalid
-        // date variable will be null, callback should expect null value for this case
-        onSelectDate(date);
-      }
-    }
-  };
-
-  private _getDefaultState(props: IDatePickerProps = this.props): IDatePickerState {
-    return {
-      selectedDate: props.value || undefined,
-      formattedDate: props.formatDate && props.value ? props.formatDate(props.value) : '',
-      isDatePickerShown: false,
-      errorMessage: undefined
-    };
-  }
-
-  private _isDateOutOfBounds(date: Date, minDate?: Date, maxDate?: Date): boolean {
-    return (!!minDate && compareDatePart(minDate!, date) > 0) || (!!maxDate && compareDatePart(maxDate!, date) < 0);
-  }
-}
+import { styled } from '../../Utilities';
+import { IDatePickerProps, IDatePickerStyleProps, IDatePickerStyles } from './DatePicker.types';
+import { DatePickerBase } from './DatePicker.base';
+import { getStyles } from './DatePicker.styles';
+
+/**
+ * DatePicker description
+ */
+export const DatePicker = styled<IDatePickerProps, IDatePickerStyleProps, IDatePickerStyles>(DatePickerBase, getStyles);
```

Comment:
might as well go ahead and add the customizer bit in on this PR `(DatePickerBase, getStyles, undefined, { scope: 'DatePicker' })`